tests/e2e/etcd_mix_versions_test.go

```diff
@@ -34,17 +34,19 @@ func TestMixVersionsSendSnapshot(t *testing.T) {
 		name               string
 		clusterVersion     e2e.ClusterVersion
 		newInstanceVersion e2e.ClusterVersion
+		injectPartition    bool
 	}{
 		// etcd doesn't support adding a new member of old version into
 		// a cluster with higher version. For example, etcd cluster
 		// version is 3.6.x, then a new member of 3.5.x can't join the
 		// cluster. Please refer to link below,
 		// https://github.com/etcd-io/etcd/blob/3e903d0b12e399519a4013c52d4635ec8bdd6863/server/etcdserver/cluster_util.go#L222-L230
-		/*{
-			name:              "etcd instance with last version receives snapshot from the leader with current version",
-			clusterVersion:    e2e.CurrentVersion,
-			newInstaceVersion: e2e.LastVersion,
-		},*/
+		{
+			name:               "etcd instance with last version receives snapshot from the leader with current version in network partition case",
+			clusterVersion:     e2e.LastVersion,
+			newInstanceVersion: e2e.CurrentVersion,
```

Comment:
This scenario has already been covered by next case (line 51~53).

I suggest to rename this test case to `TestMixVersionsSnapshotByAddingMember` and keep it as it's for now. 

We can add a new test case something like `TestMixVersionsSnapshotByMockingPartition`, and support both cases mentioned in https://github.com/etcd-io/etcd/issues/14807.  You can use [MinorityLastVersion](https://github.com/etcd-io/etcd/blob/614e520b97c72f9cfbb14cc8083e58538f05c6d0/tests/framework/e2e/cluster.go#L565-L570). In a 3 member cluster, the first member must be the current version, and the last member is previous version (e.g. 3.5.6). You can intentionally stop the first or the last member to verify both cases.

Reply:
OK, that seems a better idea, thanks !

Reply:
@ahrtr hi, I've done some changes, please have a review again, thanks !